### PR TITLE
feat: add `delete-files` flag

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain: add a new flag to automatically delete unused files

### CODE REVIEW

File: lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/unused_files_console_reporter.dart [hunk 1]
```diff
@@ -26,14 +28,18 @@ class UnusedFilesConsoleReporter
...
 
+    final removeFilesSuffix = additionalParams?.deleteUnusedFiles ?? false
+        ? ', all unused files were successfully deleted'
+        : '';
+
     output
       ..writeln('')
       ..writeln(
-        '${alarmPen('✖')} total unused files - ${alarmPen(sortedRecords.length)}',
+        '${alarmPen('✖')} total unused files - ${alarmPen(sortedRecords.length)}$removeFilesSuffix',
```
Comment (dkrutskikh) [inline lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/unused_files_console_reporter.dart:42]: @incendial if we remove all unused files, why we in alarm state? I think this is good or info state, our repo not contains unused files.

File: lib/src/analyzers/unused_files_analyzer/unused_files_analyzer.dart [hunk 2]
```diff
@@ -95,6 +96,14 @@ class UnusedFilesAnalyzer {
     }).toSet();
   }
 
+  Future<void> deleteAllUnusedFiles(
```
Comment (dkrutskikh) [inline lib/src/analyzers/unused_files_analyzer/unused_files_analyzer.dart:99]: @incendial why this is async function? I see only sync methods.

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: lib/src/analyzers/unused_files_analyzer/unused_files_analyzer.dart (hunk 2 above)
Comment (incendial) [reply lib/src/analyzers/unused_files_analyzer/unused_files_analyzer.dart]: Fixed

File: lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/unused_files_console_reporter.dart (hunk 1 above)
Comment (incendial) [reply lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/unused_files_console_reporter.dart]: Fixed

File: lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/unused_files_console_reporter.dart [hunk 3]
```diff
@@ -26,14 +28,22 @@ class UnusedFilesConsoleReporter
...
+        ..writeln(
+          '${okPen('✔')} ${alarmPen(sortedRecords.length)} files were successfully deleted',
+        );
+    } else {
+      output
+        ..writeln('')
+        ..writeln(
+          '${alarmPen('✖')} total unused files - ${alarmPen(sortedRecords.length)}$removeFilesSuffix',
+        );
+    }
```
Comment (dkrutskikh) [inline lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/unused_files_console_reporter.dart:47]: @incendial can we extract assembling log record in temporary variable?

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**